src/shared-components/alert/index.tsx

```diff
@@ -25,11 +25,24 @@ const alertIconMapping = {
   default: InfoIcon,
 };
 
-class Alert extends React.Component {
-  static Container = ({ children }) => (
-    <AlertsContainer>{children}</AlertsContainer>
-  );
+type AlertProps = {
+  avatarSrc?: string;
+  content: React.ReactNode;
+  ctaContent?: React.ReactNode;
+  duration?: string | number;
+  truncateText: boolean;
+  type: 'success' | 'error' | 'default';
```

Comment:
TS doesnt recognize defaultProps, so to avoid having to handle undefined and other verbose methods I just mark the prop as required

Reply:
This thinking is wrong, If prop is Not required then it must be marked as optional and viceversa in other words I need to honor Proptypes required/non required logic in TS

Reply:
You could try setting up a separate method like so:

```
getProps = () => {
 // Return props with return type information that includes those provided by defaultProps
}
```

And then use `const { ... } = this.getProps()` to get the type information we want. We use a similar pattern in PocketDerm for `mobx` `inject` in a few components, look up `injectedProps` in that repo. 

It would be nice to get a method that doesn't add actual code overhead though. 🙈

Reply:
No worries, just tested another solution: Making all TS prop type that have defaultProps as Not optional in TS and it works. It looks like TS is smart enough that if you use a component it wont throw an error if it has optional defaultProp and is marked as not optional in TS. this is clean and is what it is being used in the wild it seems

Reply:
I standarize the code to this solution. On the other hand the tests Im doing in test app are great, it marks the props as required/not required correctly and with intellisense end everything, this is promising

Reply:
Yep--however, this means that in consuming components, we will throw errors if those props are not provided, even though there are defaultProps present. 

I think it is actually preferable to have the low-cost conditional checks present, rather than have to update the API of all consuming components that currently rely on defaultProps.

Reply:
ok the wording is confusing, but I meant to say I did tested in consumig app (like Curology) and it does not throw an error in fact all Component Props work correctly . Looks like TS figure this out automatically (not optional but has defaultProps)

Reply:
Oh cool. Yeah maybe the types for Class components are smart enough to infer from what's provided, along with the defaultProps of the class, to figure that out. Just tested on something in the main app as well. Yay!